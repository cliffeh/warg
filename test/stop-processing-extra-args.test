#!/bin/bash

DIR=$(dirname "$0")
WARG="$DIR/../warg.test"
rc=0

color_red="$(tput setaf 1)"
color_green="$(tput setaf 2)"
color_reset="$(tput sgr0)"

EXPECTED='indent: 0
infile: -
outfile: -
prefix: 
max_frobulate: 0
verbose: 0
extra arg: foo
extra arg: bar
extra arg: baz'

if $WARG -- foo bar baz > "$0.out" 2> "$0.err"; then
  echo -n "$0: "
  if echo "$EXPECTED" | diff "$0.out" -; then
    echo -n "${color_green}PASS"
  else
    echo -n "${color_red}FAIL"
    rc=1
  fi
  echo "${color_reset}"
else
  rc=1
fi

exit $rc
