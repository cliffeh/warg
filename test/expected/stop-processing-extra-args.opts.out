indent: 0
infile: -
outfile: -
prefix: 
max_frobulate: 0
verbose: 0
extra arg: foo
extra arg: bar
extra arg: baz
